src/PrestaShopBundle/Controller/Admin/ModuleController.php

```diff
@@ -314,18 +315,173 @@ public function notificationAction()
         ));
     }
 
-    public function getPreferredModulesAction()
+    public function getPreferredModulesAction(Request $request)
     {
-        $controller = new \AdminModulesControllerCore();
-        ob_start();
+        $tabModulesList = $request->request->get('tab_modules_list');
+        $back = $request->request->get('back_tab_modules_list');
 
-        $controller->ajaxProcessGetTabModulesList();
+        if ($back) {
+            $back .= '&tab_modules_open=1';
+        }
+        if ($tabModulesList) {
+            $tabModulesList = explode(',', $tabModulesList);
+            $modulesListUnsorted = $this->getModulesByInstallation($tabModulesList, $request->request->get('admin_list_from_source'));
+        }
+
+        $installed = $uninstalled = array();
+
+        foreach ($tabModulesList as $key => $value) {
+            $continue = 0;
+            foreach ($modulesListUnsorted['installed'] as $moduleInstalled) {
+                if ($moduleInstalled->get('name') == $value) {
+                    $continue = 1;
+                    $installed[] = $moduleInstalled;
+                }
+            }
+            if ($continue) {
+                continue;
+            }
+            foreach ($modulesListUnsorted['not_installed'] as $moduleNotinstalled) {
+                if ($moduleNotinstalled->get('name') == $value) {
+                    $uninstalled[] = $moduleNotinstalled;
+                }
+            }
+        }
+
+        $moduleListSorted = array(
+            'installed' => $installed,
+            'notInstalled' => $uninstalled
+        );
+
+        $twigParams = array(
+            'currentIndex' => '',
+            'modulesList' => $moduleListSorted
+        );
+
+        if ($request->request->has('admin_list_from_source')) {
+            $twigParams['adminListFromSource'] = $request->request->get('admin_list_from_source');
+        }
+
+        return $this->render('PrestaShopBundle:Admin/Module:tab-modules-list.html.twig', $twigParams);
+    }
+
+    protected function getModulesByInstallation($modulesSelectList = null)
+    {
+        $addonsProvider = $this->get('prestashop.core.admin.data_provider.addons_interface');
+        $moduleRepository = $this->get('prestashop.core.admin.module.repository');
+        
+        $oldOnDisk = \Module::getModulesOnDisk(true, $addonsProvider->isAddonsAuthenticated(), $this->getContext()->employee->id);
+        $modulesOnDisk = $moduleRepository->getList();
+
+        $modulesList = array(
+            'installed' =>array(),
+            'not_installed' => array()
+        );
+
+        foreach ($modulesOnDisk as $module) {
+            if (!isset($modulesSelectList) || in_array($module->get('name'), $modulesSelectList)) {
+                $perm = true;
+                if ($module->get('id')) {
+                    $perm &= \Module::getPermissionStatic($module->get('id'), 'configure');
```

Comment:
\Module::getPermissionStatic already translates to the new authorization system but I think you can use
`Access::isGranted('ROLE_MOD_MODULE_'.strtoupper($module->get('name')).'_EDIT' , Context::getContext()->employee)`
or something similar, it will save a few milliseconds (: